packages/foam-core/src/markdown-provider.ts

```diff
@@ -8,26 +8,22 @@ import { Parent, Point } from 'unist';
 import * as path from 'path';
 import { NoteGraphAPI } from './note-graph';
 import { NoteLinkDefinition, Note, NoteParser } from './types';
-import { dropExtension, uriToSlug } from './utils';
+import {
+  dropExtension,
+  uriToSlug,
+  extractHashtags,
+  extractTagsFromProp,
+} from './utils';
 import { ID } from './types';
 import { ParserPlugin } from './plugins';
 
-const yamlPlugin: ParserPlugin = {
-  visit: (node, note) => {
-    if (node.type === 'yaml') {
-      note.properties = {
-        ...note.properties,
-        ...(parseYAML(node.value as string) ?? {}),
-      };
-      // Give precendence to the title from the frontmatter if it exists
-      note.title = note.properties.title ?? note.title;
-      // Update the start position of the note by exluding the metadata
-      note.source.contentStart = {
-        line: node.position!.end.line! + 1,
-        column: 1,
-        offset: node.position!.end.offset! + 1,
-      };
-    }
+const tagsPlugin: ParserPlugin = {
+  onWillVisitTree: (tree, note) => {
+    note.tags = extractHashtags(note.source.text);
+  },
+  onDidFindProperties: (props, note) => {
+    const yamlTags = extractTagsFromProp(props.tags);
```

Comment:
We can probably skip the intermediary setting of this variable, and instead chain from the function?